src/runmode-unix-socket.c

```diff
@@ -831,6 +831,74 @@ TmEcode UnixSocketDatasetLookup(json_t *cmd, json_t *answer, void *data)
     }
 }
 
+static bool GetVlanParameters(json_t *cmd, json_t *answer, TrafficId *vlan_tuple)
+{
+    memset(vlan_tuple, 0, sizeof(*vlan_tuple));
+
+    TrafficId args;
+    uint16_t count = 0;
+
+    /* 3 get optional hargs (legacy)*/
+    json_t *hargs = json_object_get(cmd, "hargs");
+    if (hargs != NULL) {
+        if (!json_is_integer(hargs)) {
+            SCLogInfo("error: hargs not a number");
+            json_object_set_new(answer, "message", json_string("hargs not a number"));
+            return false;
+        }
+        if (json_integer_value(hargs) < 1 || json_integer_value(hargs) > 4094) {
+            SCLogInfo("error: vlan ids must be between 1 and 4094");
+            json_object_set_new(answer, "message", json_string("hargs invalid vlan id"));
+            return false;
+        } else {
+            args.vlan.tuple[count++] = (uint16_t)json_integer_value(hargs);
+        }
+    }
+
+    /* 3.5 (8+ hargs_tuple Get variable-length hargs_tuple  */
+    hargs = json_object_get(cmd, "hargs_tuple");
+    if (hargs != NULL) {
+        if (json_is_array(hargs)) {
+            size_t index;
+            json_t *value;
+            json_array_foreach (hargs, index, value) {
+                int vlan = json_integer_value(value);
+                if (vlan < 0 || vlan > 4094) {
+                    SCLogInfo("error: vlan ids must be between 0 and 4094");
+                    json_object_set_new(
+                            answer, "message", json_string("hargs_tuple invalid vlan id"));
+                    return false;
+                }
+                args.vlan.tuple[count++] = (uint16_t)vlan;
+            }
+        } else if (json_is_integer(hargs)) {
+            int vlan = json_integer_value(hargs);
+            if (vlan < 0 || vlan > 4094) {
```

Comment:
0 should not be accepted, right ?

Reply:
"0" is a wildcard value

Reply:
But only for vlan_tuple, not for a single vlan, right ?